dspace-spring-rest/src/main/java/org/dspace/app/rest/submit/SubmissionService.java

```diff
@@ -0,0 +1,146 @@
+import org.dspace.app.rest.model.BitstreamRest;
+import org.dspace.app.rest.model.CheckSumRest;
+import org.dspace.app.rest.model.MetadataValueRest;
+import org.dspace.app.rest.model.ResourcePolicyRest;
+import org.dspace.app.rest.model.step.UploadBitstreamRest;
+import org.dspace.app.rest.utils.ContextUtil;
+import org.dspace.authorize.ResourcePolicy;
+import org.dspace.content.Bitstream;
+import org.dspace.content.Collection;
+import org.dspace.content.MetadataValue;
+import org.dspace.content.WorkspaceItem;
+import org.dspace.content.service.CollectionService;
+import org.dspace.content.service.WorkspaceItemService;
+import org.dspace.core.Context;
+import org.dspace.core.Utils;
+import org.dspace.services.ConfigurationService;
+import org.dspace.services.RequestService;
+import org.dspace.services.model.Request;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+/**
+ * Service to manipulate in-progress submissions.
+ * 
+ * @author Luigi Andrea Pascarelli (luigiandrea.pascarelli at 4science.it)
+ *
+ */
+@Component
+public class SubmissionService {
+
+	private static final Logger log = Logger.getLogger(SubmissionService.class);
+	
+	@Autowired
+	protected ConfigurationService configurationService;
+	@Autowired
+	protected CollectionService collectionService;
+	@Autowired
+	protected WorkspaceItemService workspaceItemService;
+	@Autowired
+	private RequestService requestService;
+	@Autowired(required = true)
+	BitstreamFormatConverter bfConverter;
+	@Autowired(required = true)
+	ResourcePolicyConverter aCConverter; 
+	
+	public WorkspaceItem createWorkspaceItem(Context context, Request request) {
+		WorkspaceItem wsi = null;
+		String collectionUUID = request.getHttpServletRequest().getParameter("collection");
+		if(StringUtils.isBlank(collectionUUID)) 
+		{
+			String uuid = configurationService.getProperty("submission.default.collection");
+			Collection collection = null;
+			try {
+			if(StringUtils.isNotBlank(uuid)) {
+				collection = collectionService.find(context, UUID.fromString(uuid));
+			}
+			else {
+				collection = collectionService.findAll(context, 1, 0).get(0);
+			}
+			wsi = workspaceItemService.create(context, collection, true);
+			} catch (Exception e) {
+				log.error(e.getMessage(), e);
+			}
+		}
+		else {
+			//TODO manage setup of default collection in the case WSI it is not null
+			//TODO manage setup of collection discovered into request
+		}
+		return wsi;
+	}
+	
+	public void saveWorkspaceItem(Context context, WorkspaceItem wsi) {
+		try {
+			workspaceItemService.update(context, wsi);
+		} catch (Exception e) {
+			log.error(e.getMessage(), e);
+		}
+	}
+	
+
+	public UploadBitstreamRest buildUploadBitstream(ConfigurationService configurationService, Bitstream source) throws SQLException {
+		UploadBitstreamRest data = new UploadBitstreamRest();
+		
+		for (MetadataValue md : source.getMetadata()) {
```

Comment:
This feels like code that should go to a `UploadBitstreamRestConverter` class similar to how all other REST models are built in a converter class.

Reply:
I agreed with @tomdesair but investigated in depth I left this code outside a "converter" because at moment there are code that it is not initialized as spring bean (see org.dspace.app.rest.submit.step.UploadStep). The converter will be available after the submission reader enhancement (i.e. the element into item-submission.xml will defined as spring bean)